File: examples/plugin-py/handler.py
import flatbuffers
from fb.echo import EchoResponse
from pluginart_helpers import BuildEchoCallResponse, DecodeEchoRequest


def handle(payload: bytes) -> bytes:
    echo_req, call = DecodeEchoRequest(payload)
    output = (echo_req.Input() or b'').decode('utf-8').upper()

    b = flatbuffers.Builder(128)
    out_off = b.CreateString(output)
    EchoResponse.EchoResponseStart(b)
    EchoResponse.EchoResponseAddOutput(b, out_off)
    echo_resp_off = EchoResponse.EchoResponseEnd(b)
    return BuildEchoCallResponse(call, b, echo_resp_off)

File: examples/plugin-ts/handler.ts
import * as flatbuffers from 'flatbuffers';
import { EchoResponse } from './plugin/fb/echo/echo-response';
import { BuildEchoCallResponse, DecodeEchoRequest } from './plugin/pluginart_helpers';

export function handle(payload: Buffer): Buffer {
  const request = DecodeEchoRequest(payload);
  const input = request.payload.input() ?? '';
  const output = input.toUpperCase();

  const builder = new flatbuffers.Builder(128);
  const outputOffset = builder.createString(output);
  EchoResponse.startEchoResponse(builder);
  EchoResponse.addOutput(builder, outputOffset);
  const responseOffset = EchoResponse.endEchoResponse(builder);
  return BuildEchoCallResponse(request.call, builder, responseOffset);
}
